advisor/src/main/kotlin/Advisor.kt

```diff
@@ -45,14 +44,10 @@ class Advisor(
     private val config: AdvisorConfiguration
 ) {
     companion object : Logging {
-        private val LOADER = ServiceLoader.load(AdviceProviderFactory::class.java)
-
         /**
          * All [advice provider factories][AdviceProviderFactory] available in the classpath, associated by their names.
          */
-        val ALL: SortedMap<String, AdviceProviderFactory> by lazy {
-            LOADER.iterator().asSequence().associateByTo(sortedMapOf(String.CASE_INSENSITIVE_ORDER)) { it.name }
-        }
+        val ALL = NamedPlugin.getAll<AdviceProviderFactory>()
```

Comment:
commit-msg: Please clarify the intention of removing the `lazy` execution.

Reply:
The removal of `lazy` was actually an oversight; I've added it back.